src/policy/concrete.rs

```diff
@@ -668,19 +668,13 @@ impl<Pk: MiniscriptKey> Policy<Pk> {
 
     /// Gets all keys in the policy.
     pub fn keys(&self) -> Vec<&Pk> {
-        match *self {
-            Policy::Key(ref pk) => vec![pk],
-            Policy::Threshold(_k, ref subs) => {
-                subs.iter().flat_map(|sub| sub.keys()).collect::<Vec<_>>()
+        let mut keys = vec![];
+        for data in self.post_order_iter() {
```

Comment:
In 062e4009c36126e9cf39117170747bcb5cdc2aa5:

I think we should use the pre-order iterator here, which is more intuitively correct. (In general, the post-order iterator matches recursive logic, but I think that, since we're only counting leaf nodes, the two iterators will actually have the same effect.)

Reply:
Oh my bad, I had it in my head that the post order iter was the simpler of the two, I was wrong. Defaulting to pre-order iter when we don't need references to the child nodes.

Reply:
Well, post-order is definitely the most *common* because it allows you to simulate depth-first recursive algorithms. But conceptually I think pre-order is more natural.